### PR TITLE
Modify GHA CI to use PYTORCH_IGNORE_DISABLED_ISSUES based on PR body

### LINKED ISSUE #60130: DISABLED test_ddp_logging_data_cpu (__main__.TestDistBackendWithFork)
## 🐛 Bug

Example failures:
https://app.circleci.com/pipelines/github/pytorch/pytorch/337673/workflows/db188d30-2c92-4faf-ac2a-e90b70d1e8c2/jobs/14187632
https://app.circleci.com/pipelines/github/pytorch/pytorch/337701/workflows/6e6161a7-7346-403a-a458-97f5e7970279/jobs/14188746
https://app.circleci.com/pipelines/github/pytorch/pytorch/337710/workflows/3f835c0e-ad07-45fd-9ede-dec90c043bb3/jobs/14188742

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu @gcramer23

### LINKED ISSUE #61655: DISABLED test_jit_cuda_extension (__main__.TestCppExtensionJIT)
See: https://github.com/pytorch/pytorch/runs/3065374723
platforms: windows

```
======================================================================
ERROR [20.716s]: test_jit_cuda_extension (__main__.TestCppExtensionJIT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_cpp_extensions_jit.py", line 109, in test_jit_cuda_extension
    z = module.sigmoid_add(x, y).cpu()
RuntimeError: CUDA error: the provided PTX was compiled with an unsupported toolchain.
CUDA kernel errors might be asynchronously reported at some other API call,so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.

----------------------------------------------------------------------
```

also trying out the new platform DISABLED capability in #61427

cc @ezyang @gchanan @zou3519 @peterjc123 @mszhanyi @skyline75489 @nbcsm @malfet @bdhirsh @jbschlosser @anjali411 @gmagogsfm

### LINKED ISSUE #61670: DISABLED test_shutdown_errors (__main__.TensorPipeRpcTestWithSpawn)
https://app.circleci.com/pipelines/github/pytorch/pytorch/350293/workflows/f3cd8e15-1a28-4ce6-9522-f5d1cf4a6c86/jobs/14756009/steps

```
Jul 13 21:24:24 ======================================================================
Jul 13 21:24:24 ERROR [2.532s]: test_shutdown_errors (__main__.TensorPipeRpcTestWithSpawn)
Jul 13 21:24:24 ----------------------------------------------------------------------
Jul 13 21:24:24 Traceback (most recent call last):
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 398, in wrapper
Jul 13 21:24:24     self._join_processes(fn)
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 597, in _join_processes
Jul 13 21:24:24     self._check_return_codes(elapsed_time)
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 637, in _check_return_codes
Jul 13 21:24:24     raise RuntimeError(error)
Jul 13 21:24:24 RuntimeError: Process 2 exited with error code 10 and exception:
Jul 13 21:24:24 RuntimeError: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:24:24 
Jul 13 21:24:24 During handling of the above exception, another exception occurred:
Jul 13 21:24:24 
Jul 13 21:24:24 Traceback (most recent call last):
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 516, in run_test
Jul 13 21:24:24     getattr(self, test_name)()
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 400, in wrapper
Jul 13 21:24:24     fn()
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 100, in new_test_method
Jul 13 21:24:24     return_value = old_test_method(self, *arg, **kwargs)
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4101, in test_shutdown_errors
Jul 13 21:24:24     rpc.shutdown()
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/unittest/case.py", line 217, in __exit__
Jul 13 21:24:24     expected_regex.pattern, str(exc_value)))
Jul 13 21:24:24   File "/opt/conda/lib/python3.6/unittest/case.py", line 135, in _raiseFailure
Jul 13 21:24:24     raise self.test_case.failureException(msg)
Jul 13 21:24:24 AssertionError: "simulation rref" does not match "eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)"
```

```
Jul 13 21:22:33   test_shutdown_errors (__main__.TensorPipeRpcTestWithSpawn) ... [W tensorpipe_agent.cpp:608] RPC agent for worker1 won't send response to request #3 to worker0, as the agent is shutting down
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker1 encountered error when reading incoming request from worker0: pipe closed (this error originated at tensorpipe/core/pipe_impl.cc:356)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:912] RPC agent for worker0 encountered error when reading incoming response from worker1: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker0 encountered error when reading incoming request from worker1: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker3 encountered error when reading incoming request from worker0: pipe closed (this error originated at tensorpipe/core/pipe_impl.cc:356)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker0 encountered error when reading incoming request from worker3: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 Failed to respond to 'Shutdown Proceed' in time, got error Followers ['worker1', 'worker2', 'worker3'] timed out in _all_gather after 5.00 seconds. The first exception is eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:608] RPC agent for worker0 won't send response to request #1 to worker2, as the agent is shutting down
Jul 13 21:22:33 [W tensorpipe_agent.cpp:912] RPC agent for worker2 encountered error when reading incoming response from worker0: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker2 encountered error when reading incoming request from worker0: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 [W tensorpipe_agent.cpp:707] RPC agent for worker0 encountered error when reading incoming request from worker2: pipe closed (this error originated at tensorpipe/core/pipe_impl.cc:356)
Jul 13 21:22:33 On WorkerInfo(id=2, name=worker2):
Jul 13 21:22:33 RuntimeError('simulation',)
Jul 13 21:22:33 Traceback (most recent call last):
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/internal.py", line 204, in _run_function
Jul 13 21:22:33     result = python_udf.func(*python_udf.args, **python_udf.kwargs)
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4090, in raise_error
Jul 13 21:22:33     raise RuntimeError('simulation')
Jul 13 21:22:33 RuntimeError: simulation
Jul 13 21:22:33 
Jul 13 21:22:33 Failed to respond to 'Shutdown Proceed' in time, got error eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 Caught exception: 
Jul 13 21:22:33 RuntimeError: eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)
Jul 13 21:22:33 
Jul 13 21:22:33 During handling of the above exception, another exception occurred:
Jul 13 21:22:33 
Jul 13 21:22:33 Traceback (most recent call last):
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 516, in run_test
Jul 13 21:22:33     getattr(self, test_name)()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 400, in wrapper
Jul 13 21:22:33     fn()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 100, in new_test_method
Jul 13 21:22:33     return_value = old_test_method(self, *arg, **kwargs)
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4101, in test_shutdown_errors
Jul 13 21:22:33     rpc.shutdown()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/unittest/case.py", line 217, in __exit__
Jul 13 21:22:33     expected_regex.pattern, str(exc_value)))
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/unittest/case.py", line 135, in _raiseFailure
Jul 13 21:22:33     raise self.test_case.failureException(msg)
Jul 13 21:22:33 AssertionError: "simulation rref" does not match "eof (this error originated at tensorpipe/transport/shm/connection_impl.cc:259)"
Jul 13 21:22:33  exiting process 2 with exit code: 10
Jul 13 21:22:33 Caught exception: 
Jul 13 21:22:33 Traceback (most recent call last):
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 516, in run_test
Jul 13 21:22:33     getattr(self, test_name)()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 400, in wrapper
Jul 13 21:22:33     fn()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 100, in new_test_method
Jul 13 21:22:33     return_value = old_test_method(self, *arg, **kwargs)
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4109, in test_shutdown_errors
Jul 13 21:22:33     dist.barrier()
Jul 13 21:22:33   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 2617, in barrier
Jul 13 21:22:33     work.wait()
Jul 13 21:22:33 RuntimeError: [/var/lib/jenkins/workspace/third_party/gloo/gloo/transport/tcp/pair.cc:598] Connection closed by peer [172.17.0.2]:14727
Jul 13 21:22:33  exiting process 0 with exit code: 10
Jul 13 21:22:34 On WorkerInfo(id=1, name=worker1):
Jul 13 21:22:34 RuntimeError('simulation',)
Jul 13 21:22:34 Traceback (most recent call last):
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/internal.py", line 204, in _run_function
Jul 13 21:22:34     result = python_udf.func(*python_udf.args, **python_udf.kwargs)
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4090, in raise_error
Jul 13 21:22:34     raise RuntimeError('simulation')
Jul 13 21:22:34 RuntimeError: simulation
Jul 13 21:22:34 
Jul 13 21:22:34 Caught exception: 
Jul 13 21:22:34 Traceback (most recent call last):
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 516, in run_test
Jul 13 21:22:34     getattr(self, test_name)()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 400, in wrapper
Jul 13 21:22:34     fn()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 100, in new_test_method
Jul 13 21:22:34     return_value = old_test_method(self, *arg, **kwargs)
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4109, in test_shutdown_errors
Jul 13 21:22:34     dist.barrier()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 2617, in barrier
Jul 13 21:22:34     work.wait()
Jul 13 21:22:34 RuntimeError: [/var/lib/jenkins/workspace/third_party/gloo/gloo/transport/tcp/pair.cc:598] Connection closed by peer [172.17.0.2]:43947
Jul 13 21:22:34  exiting process 1 with exit code: 10
Jul 13 21:22:34 On WorkerInfo(id=3, name=worker3):
Jul 13 21:22:34 RuntimeError('simulation',)
Jul 13 21:22:34 Traceback (most recent call last):
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/rpc/internal.py", line 204, in _run_function
Jul 13 21:22:34     result = python_udf.func(*python_udf.args, **python_udf.kwargs)
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4090, in raise_error
Jul 13 21:22:34     raise RuntimeError('simulation')
Jul 13 21:22:34 RuntimeError: simulation
Jul 13 21:22:34 
Jul 13 21:22:34 Caught exception: 
Jul 13 21:22:34 Traceback (most recent call last):
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 516, in run_test
Jul 13 21:22:34     getattr(self, test_name)()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 400, in wrapper
Jul 13 21:22:34     fn()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 100, in new_test_method
Jul 13 21:22:34     return_value = old_test_method(self, *arg, **kwargs)
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 4109, in test_shutdown_errors
Jul 13 21:22:34     dist.barrier()
Jul 13 21:22:34   File "/opt/conda/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 2617, in barrier
Jul 13 21:22:34     work.wait()
Jul 13 21:22:34 RuntimeError: [/var/lib/jenkins/workspace/third_party/gloo/gloo/transport/tcp/pair.cc:598] Connection closed by peer [172.17.0.2]:16786
Jul 13 21:22:34  exiting process 3 with exit code: 10
Jul 13 21:22:34 Process 2 terminated with exit code 10, terminating remaining processes.
Jul 13 21:22:34 ERROR (2.532s)
```

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @jjlilley @mrzzd @agolynski @cbalioglu @gcramer23

### PR DESCRIPTION
**Pull request by janeyx99**
Another step forward in fixing https://github.com/pytorch/pytorch/issues/62359

Disclaimer: this only works with GHA for now, as circleci would require changes in probot.

Test plan can be seen a previous description where I modified the description to include linked issues. I've removed them now since the actual PR doesn't fix any of them. 

It works! In the [periodic 11.3 test1](https://github.com/pytorch/pytorch/pull/62851/checks?check_run_id=3263109970), we get this in the logs and we see that PYTORCH_IGNORE_DISABLED_ISSUES is properly set:
```
  test_jit_cuda_extension (__main__.TestCppExtensionJIT) ... Using /var/lib/jenkins/.cache/torch_extensions/py36_cu113 as PyTorch extensions root...
Creating extension directory /var/lib/jenkins/.cache/torch_extensions/py36_cu113/torch_test_cuda_extension...
Detected CUDA files, patching ldflags
Emitting ninja build file /var/lib/jenkins/.cache/torch_extensions/py36_cu113/torch_test_cuda_extension/build.ninja...
Building extension module torch_test_cuda_extension...
Using envvar MAX_JOBS (30) as the number of workers...
[1/3] c++ -MMD -MF cuda_extension.o.d -DTORCH_EXTENSION_NAME=torch_test_cuda_extension -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /opt/conda/lib/python3.6/site-packages/torch/include -isystem /opt/conda/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.6/site-packages/torch/include/TH -isystem /opt/conda/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /opt/conda/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=1 -fPIC -std=c++14 -c /var/lib/jenkins/workspace/test/cpp_extensions/cuda_extension.cpp -o cuda_extension.o 
[2/3] /usr/local/cuda/bin/nvcc  -DTORCH_EXTENSION_NAME=torch_test_cuda_extension -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -isystem /opt/conda/lib/python3.6/site-packages/torch/include -isystem /opt/conda/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.6/site-packages/torch/include/TH -isystem /opt/conda/lib/python3.6/site-packages/torch/include/THC -isystem /usr/local/cuda/include -isystem /opt/conda/include/python3.6m -D_GLIBCXX_USE_CXX11_ABI=1 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_50,code=sm_50 -gencode=arch=compute_52,code=compute_52 -gencode=arch=compute_52,code=sm_52 --compiler-options '-fPIC' -O2 -std=c++14 -c /var/lib/jenkins/workspace/test/cpp_extensions/cuda_extension.cu -o cuda_extension.cuda.o 
nvcc warning : The 'compute_35', 'compute_37', 'compute_50', 'sm_35', 'sm_37' and 'sm_50' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
[3/3] c++ cuda_extension.o cuda_extension.cuda.o -shared -L/opt/conda/lib/python3.6/site-packages/torch/lib -lc10 -lc10_cuda -ltorch_cpu -ltorch_cuda_cu -ltorch_cuda_cpp -ltorch -ltorch_python -L/usr/local/cuda/lib64 -lcudart -o torch_test_cuda_extension.so
Loading extension module torch_test_cuda_extension...
ok (26.161s)
```

whereas on the latest master periodic 11.1 windows [test](https://github.com/pytorch/pytorch/runs/3263762478?check_suite_focus=true), we see
```
test_jit_cuda_extension (__main__.TestCppExtensionJIT) ... skip (0.000s)
```

### CODE REVIEW

File: .github/scripts/generate_pytorch_test_matrix.py [hunk 1]
```diff
@@ -19,6 +20,13 @@ class Config(TypedDict):
     runner: str
 
 
+def get_disabled_issues() -> str:
+    pr_body = os.getenv('PR_BODY', '')
+    regex = '(?i)(Close(d|s)?|Resolve(d|s)?|Fix(ed|es)?) #([0-9]+)'
```
Comment (zhouzhuojie) [inline .github/scripts/generate_pytorch_test_matrix.py:25]: can we have either comments or unit tests to demo some of the valid/invalid cases?

Comment (janeyx99) [reply .github/scripts/generate_pytorch_test_matrix.py:25]: Ah yes I can definitely add some comments

Comment (walterddr) [reply .github/scripts/generate_pytorch_test_matrix.py:25]: FYI: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Comment (janeyx99) [reply .github/scripts/generate_pytorch_test_matrix.py]: Yes, that's what I based it on. I added a comment with that link just now.

**Review by walterddr — APPROVED** [review]

**Pull request closed without merging**